### PR TITLE
Adapt code coverage to Go 1.20

### PR DESCRIPTION
**Pull request by adrianreber**
This will not work with older versions of Go.

This also introduces a commit to actually run the tests during CI. `test` -> `./test`.

### CODE REVIEW

**Comment by adrianreber** [comment]
Now coverage works again, but the coverage is really low. Hmm. Ah, I might need to exclude certain packages.

File: test/Makefile [hunk 1]
```diff
@@ -41,41 +41,38 @@ crit-test:
 	$(MAKE) -C crit
 
 test.coverage: *.go
-	$(GO) test \
-		-covermode=count \
-		-coverpkg=./... \
-		-mod=vendor \
-		-tags coverage \
-		-buildmode=pie -c -o $@ $^
+	# This requires at least Go 1.20
```
Comment (rst0git) [inline test/Makefile:44]: Would it be better to use the following to check the Go version?

```Makefile
GO_MAJOR_VER = $(shell $(GO) version | cut -c 14- | cut -d' ' -f1 | cut -d'.' -f1)
GO_MINOR_VER = $(shell $(GO) version | cut -c 14- | cut -d' ' -f1 | cut -d'.' -f2)
MIN_GO_MAJOR_VER = 1
MIN_GO_MINOR_VER = 20
GO_VALIDATION_ERR = Go version is not supported. Please update to at least $(MIN_GO_MAJOR_VER).$(MIN_GO_MINOR_VER)

check-go-version:
	@if [ $(GO_MAJOR_VER) -gt $(MIN_GO_MAJOR_VER) ]; then \
		exit 0 ;\
	elif [ $(GO_MAJOR_VER) -lt $(MIN_GO_MAJOR_VER) ]; then \
		echo '$(GO_VALIDATION_ERR)';\
		exit 1; \
	elif [ $(GO_MINOR_VER) -lt $(MIN_GO_MINOR_VER) ] ; then \
		echo '$(GO_VALIDATION_ERR)';\
		exit 1; \
	fi

.PHONY: check-go-version
```
```Makefile
coverage: check-go-version $(COVERAGE_BINARIES) $(TEST_PAYLOAD)
```

**Comment by adrianreber** [comment]
image

The problem was that the whole setup was broken. We were only collecting coverage from the `test/` directory. So the code coverage setup is correct.

File: test/Makefile (hunk 1 above)
Comment (adrianreber) [reply test/Makefile:44]: good idea, will do that

Comment (adrianreber) [reply test/Makefile]: done

**Comment by adrianreber** [comment]
I think this is ready now. The drop in test coverage is correct, because it was just plainly wrong what was reported before. Now we get the test coverage from all our code before we just got the test coverage from the test binaries (which is a pretty useless metric).

**Pull request merged**